### ISSUE TITLE
Disassemble regex in 930100

### ISSUE THREAD

**Issue by dune73** [coreruleset/coreruleset#2221]
The regex in 930100 is rather hard to read. It would be worthwhile to disassemble it and then generate it via script:

```
(?i)(?:\x5c|(?:%(?:c(?:0%(?:[2aq]f|5c|9v)|1%(?:[19p]c|8s|af))|2(?:5(?:c(?:0%25af|1%259c)|2f|5c)|%46|f)|(?:(?:f(?:8%8)?0%8|e)0%80%a|bg%q)f|%3(?:2(?:%(?:%6|4)6|F)|5%%63)|u(?:221[56]|002f|EFC8|F025)|1u|5c)|0x(?:2f|5c)|/))(?:%(?:(?:f(?:(?:c%80|8)%8)?0%8|e)0%80%ae|2(?:(?:5(?:c0%25a|2))?e|%45)|u(?:(?:002|ff0)e|2024)|%32(?:%(?:%6|4)5|E)|c0(?:%[256aef]e|\.))|\.(?:%0[01]|\?)?|\?\.?|0x2e){2}(?:\x5c|(?:%(?:c(?:0%(?:[2aq]f|5c|9v)|1%(?:[19p]c|8s|af))|2(?:5(?:c(?:0%25af|1%259c)|2f|5c)|%46|f)|(?:(?:f(?:8%8)?0%8|e)0%80%a|bg%q)f|%3(?:2(?:%(?:%6|4)6|F)|5%%63)|u(?:221[56]|002f|EFC8|F025)|1u|5c)|0x(?:2f|5c)|/))
```

For the record: 930100 protects us from CVE-2021-41773, yet looking at the regex I can not readily see how.

**Comment by franbuehler** [comment]
Hi @dune73,
This seems to be a task for me??

**Comment by dune73** [comment]
If you think that would be a fun adventure, then that would be most welcome. :)

**Comment by theseion** [comment]
@franbuehler would it make sense to write a script to simplify such expressions into palatable subexpressions? How do you usually decompose these?

**Comment by franbuehler** [comment]
The regex has probably been generated with Regexp::Assemble, but the problem is that the source is no avaliable anymore.
Therefore I disassembled all SQLi rules and wrote the source files/data files that we have today.
I did this manually because there were too many special cases or I was just not capable to write a script ;)

How I did this is written in this blog post here:
https://coreruleset.org/20171109/disassembling-sqli-rules/

**Comment by dune73** [comment]
CRS lore also has it that @franbuehler also received an admission into the ranks of the "International Academy for Advanced Regular Expression Wizardry" aka as `/^IA{2}REW$/` for this work.

**Comment by theseion** [comment]
That's great! Thanks @RedXanadu. With the extended assemble script we can use prefix and suffix markers to generate this.

**Comment by RedXanadu** [comment]
I'm not 100% sure the new regex script can completely replicate this regular expression. I think some new, additional comments will need to be written to accompany the rule.

I'll have a look at this later this week.

**Comment by theseion** [comment]
You're probably right. I'm working on another preprocessor that should make that possible.

**Comment by RedXanadu** [comment]
I think this is resolved by #2298, which I believe is ready to go.

**Comment by RedXanadu** [comment]
Resolved by #2298, which has now been Approved and Merged.
